.github/workflows/build-and-test.yaml

```diff
@@ -30,21 +38,56 @@ jobs:
         run: echo $CONFIG_E2E_CLOUD > FRTestHost/FRTestHost/SharedTestFiles/TestConfig/Config-live-01.json
 
       # Set target Xcode version. For more details and options see:
-      # https://github.com/actions/virtual-environments/blob/main/images/macos/macos-12-Readme.md
+      # https://github.com/actions/runner-images/blob/main/images/macos/macos-15-Readme.md
       - name: Select Xcode
-        run: sudo xcode-select -switch /Applications/Xcode_15.4.app && /usr/bin/xcodebuild -version
+        run: sudo xcode-select -switch /Applications/Xcode_${{ env.XCODE_VERSION }}.app && /usr/bin/xcodebuild -version
+
+      # # Restore `xcresultparser` from cache
+      # - name: Restore xcresultparser from cache
+      #   id: cache-xcresultparser
+      #   uses: actions/cache@v3
+      #   with:
+      #     path: /opt/homebrew/bin/xcresultparser
+      #     key: ${{ runner.os }}-xcresultparser
+      #     restore-keys: |
+      #       ${{ runner.os }}-xcresultparser
+
+      # Install xcresultparser
+      # TBD: Cahce xcresultparser tool. It is not working as expected at the moment, causing the build to fail...
+      - name: Install xcresultparser
+        run: |
+          brew tap a7ex/homebrew-formulae
+          brew install xcresultparser
+          echo "Installed xcresultparser..."
 
       # Run all tests
       - name: Run tests
-        run: xcodebuild test -scheme FRTestHost -workspace e2e/FRExample.xcworkspace -configuration Debug -destination 'platform=iOS Simulator,name=iPhone 14,OS=17.5' -derivedDataPath DerivedData -enableCodeCoverage YES -resultBundlePath TestResults | xcpretty && exit ${PIPESTATUS[0]}
+        run: |
+          xcodebuild test \
+          -scheme FRTestHost \
+          -workspace ${{ env.WORKSPACE }} \
+          -configuration ${{ env.CONFIGURATION}} \
+          -destination '${{ env.DESTINATION }}' \
+          -derivedDataPath DerivedData \
+          TEST_TARGET_SIGNING=YES \
+          -enableCodeCoverage YES \
+          -resultBundlePath TestResults | grep -E 'Test Case|^Executed|error:'
 
-      # Publish test results
+      # Convert test results to JUnit format
+      - name: Convert Test Results to JUnit
+        run: xcresultparser -o junit TestResults.xcresult > test-report.xml
+
+      # Publish the test results
       - name: Publish test results
-        uses: kishikawakatsumi/xcresulttool@v1
```

Comment:
Unfortunately this `xcresulttool` action is not working anymore. Causing more troubles than good. Switched to more reliable `dorny/test-reporter` instead...